pkg/config/derive.go

```diff
@@ -12,24 +13,55 @@ type derived struct{}
 
 var Derived derived
 
-func (derived) SystemReserved(cs *api.OpenShiftManagedCluster, role api.AgentPoolProfileRole) string {
-	for _, pool := range cs.Properties.AgentPoolProfiles {
-		if pool.Role != role {
+func isSmallVM(vmSize api.VMSize) bool {
+	// TODO: we should only be allowing StandardD2sV3 for test
+	return vmSize == api.StandardD2sV3
+}
+
+func (derived) SystemReserved(cs *api.OpenShiftManagedCluster, role api.AgentPoolProfileRole) (string, error) {
+	for _, app := range cs.Properties.AgentPoolProfiles {
+		if app.Role != role {
 			continue
 		}
-		return api.DefaultVMSizeKubeArguments[pool.VMSize][role][api.SystemReserved]
+
+		if isSmallVM(app.VMSize) {
+			if role == api.AgentPoolProfileRoleMaster {
+				return "cpu=500m,memory=1Gi", nil
+			} else {
+				return "cpu=200m,memory=512Mi", nil
+			}
+
+		} else {
+			if role == api.AgentPoolProfileRoleMaster {
+				return "cpu=1000m,memory=1Gi", nil
+			} else {
+				return "cpu=500m,memory=512Mi", nil
+			}
+		}
 	}
-	return ""
+
+	return "", fmt.Errorf("role %s not found", role)
 }
 
-func (derived) KubeReserved(cs *api.OpenShiftManagedCluster, role api.AgentPoolProfileRole) string {
-	for _, pool := range cs.Properties.AgentPoolProfiles {
-		if pool.Role != role {
+func (derived) KubeReserved(cs *api.OpenShiftManagedCluster, role api.AgentPoolProfileRole) (string, error) {
+	if role == api.AgentPoolProfileRoleMaster {
+		return "", fmt.Errorf("kubereserved not defined for role %s", role)
```

Comment:
nit: this error message sounds like you aren't setting kubereserved for the master role, when it should convey the opposite, change to something like "cannot set kubereserved for role %s" or similar

Reply:
if this error appears, it is indicative of a programming error, not an end-user one.